api/src/main/java/org/openmrs/module/appframework/service/AppFrameworkServiceImpl.java

```diff
@@ -421,5 +422,16 @@ public void purgeUserApp(UserApp userApp) {
         new AppFrameworkActivator().contextRefreshed();
     }
 
+	@Override
+	public List<Extension> getAllExtensions() {
+
+		List<Extension> allExtensions = new ArrayList<Extension>();
+
+		for (Extension extension : allFreeStandingExtensions.getExtensions()) {
```

Comment:
What are we accomplishing by simply looping through a collection to create another?

Reply:
I looped through so as to add the extensions to the list and be able to return a list

Reply:
Isn't it already provided by this? allFreeStandingExtensions.getExtensions()